src/IonEventStream.ts

```diff
@@ -0,0 +1,410 @@
+        for(tid = this.reader.next(); tid !== undefined; tid = this.reader.next()) {
+
+            let fieldName = this.reader.fieldName();
+            if (currentEvent.has(fieldName)) {
+                throw new Error('Repeated event field: ' + fieldName);
+            }
+            switch (fieldName) {
+                case 'event_type' : {
+                    currentEvent.set(fieldName, this.reader.stringValue());
+                    break;
+                }
+
+                case 'ion_type' : {
+                    currentEvent.set(fieldName, this.parseIonType());
+                    break;
+                }
+
+                case 'field_name' : {
+                    currentEvent.set(fieldName, this.reader.stringValue());
+                    break;
+                }
+
+                case 'annotations' : {
+                    currentEvent.set(fieldName, this.parseAnnotations());
+                    break;
+                }
+
+                case 'value_text' : {
+                    let tempString : string = this.reader.stringValue();
+                    let tempReader : Reader = makeReader(tempString, undefined);
+                    tempReader.next();
+                    currentEvent.set('isNull', tempReader.isNull());
+                    currentEvent.set(fieldName,tempReader.value());
+                    break;
+                }
+
+                case 'value_binary' : {
+                    currentEvent.set(fieldName, this.parseBinaryValue());
+                    break;
+                }
+
+                case 'imports' : {
+                    currentEvent.set(fieldName, this.parseImports());
+                    break;
+                }
+
+                case 'depth' : {
+                    currentEvent.set(fieldName, this.reader.numberValue());
+                    break;
+                }
+
+                default :
+                    throw new Error('Unexpected event field: ' + fieldName);
+            }
+        }
+        let eventType : IonEventType;
+        switch(currentEvent.get('event_type')){
+            case 'CONTAINER_START' :
+                eventType = IonEventType.CONTAINER_START;
+                break;
+            case 'STREAM_END' :
+                eventType = IonEventType.STREAM_END;
+                break;
+            case 'CONTAINER_END' :
+                eventType = IonEventType.CONTAINER_END;
+                break;
+            case 'SCALAR' :
+                eventType = IonEventType.SCALAR;
+                break;
+            case 'SYMBOL_TABLE' :
+                throw new Error('Symbol tables unsupported');
+        }
+        return this.eventFactory.makeEvent(
+            eventType,
+            currentEvent.get('ion_type'),
+            currentEvent.get('field_name'),
+            currentEvent.get('depth'),
+            currentEvent.get('annotations'),
+            currentEvent.get('isNull'),
+            currentEvent.get('value_text')
```

Comment:
Need to verify that value_text and value_binary are data model equivalent. Otherwise, it's an invalid event and an error should be raised.

Reply:
Binary was blowing everything up so i killed it until work starts on that side.

Reply:
Sure. Might consider adding a placeholder for the validation logic, because it will be important to add once there's binary support.

Reply:
https://github.com/amzn/ion-js/issues/131